client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpClient.ts

```diff
@@ -34,19 +34,11 @@ export class HttpClient implements IHttpClient {
                     resolve(xhr.response);
                 }
                 else {
-                    reject({
-                        status: xhr.status,
-                        statusText: xhr.statusText
-                    });
+                    reject(new Error(xhr.statusText));
```

Comment:
With this change you are losing status code. We need both. I would add it to the message details. There might be value in storing the status code in a property as well.